src/shared-components/accordion/index.tsx

```diff
@@ -32,131 +32,97 @@ type AccordionProps = {
  *
  * The accordion component expands to reveal hidden information. They should be used when you need to fit a large amount of content but don't want to visually overwhelm the user.
  */
-export class Accordion extends React.Component<
-  AccordionProps & Required<AccordionProps>,
-  { contentHeight: string }
-> {
-  static propTypes = {
-    /** Sets the border-radius of Accordion.Container, AccordionBox, and TitleWrapper */
-    borderRadius: PropTypes.string,
-    /** node(s) that will render only when expanded */
-    children: PropTypes.node.isRequired,
-    /** when true, the accordion will be greyed out and the onClick prop will be disabled */
-    disabled: PropTypes.bool,
-    /** determine if the accordion is collapsed (false) or expanded (true) */
-    isOpen: PropTypes.bool.isRequired,
-    /** when true, border lines between accordions and title/children nodes will disappear */
-    noBorder: PropTypes.bool,
-    /** invoked when title node is clicked */
-    onClick: PropTypes.func.isRequired,
-    /** when true, the arrow is aligned flush with the right side of the component */
-    rightAlignArrow: PropTypes.bool,
-    /** node that will render whether collapsed or expanded */
-    title: PropTypes.node.isRequired,
-  };
-
-  static defaultProps = {
-    borderRadius: '4px',
-    disabled: false,
-    noBorder: false,
-    rightAlignArrow: false,
-  };
-
-  static Container = Container;
-
-  static Content = Content;
-
-  static Thumbnails = Thumbnails;
-
-  static Truncate = Truncate;
-
-  contentRef = React.createRef<HTMLDivElement>();
-
-  state = { contentHeight: '0px' };
-
-  componentDidMount() {
-    this.updateHeight();
-  }
-
-  componentDidUpdate() {
-    this.updateHeight();
-  }
-
-  // prettier-ignore
-  getContentHeight = (isOpen: boolean) => (
-    `${
-      isOpen && this.contentRef.current
-        ? this.contentRef.current.clientHeight
-        : '0'
-    }px`
-  );
-
-  updateHeight() {
-    const { isOpen } = this.props;
-    const { contentHeight } = this.state;
-
-    const nextHeight = this.getContentHeight(isOpen);
+export const Accordion = ({
+  borderRadius = '4px',
+  children,
+  disabled = false,
+  isOpen,
+  noBorder = false,
+  onClick,
+  rightAlignArrow = false,
+  title,
+}: AccordionProps) => {
+  const [contentHeight, setContentHeight] = useState('0px');
+
+  const contentRef = React.createRef<HTMLDivElement>();
+
+  useEffect(() => {
+    const nextHeight = `${
```

Comment:
It's less DRY but I think we should just make this more readable instead of continuing to use what was present before:

```tsx
const nextHeight = isOpen && contentRef.current ? `${contentRef.current.clientHeight}px` : '0px'
```

Reply:
done